lips/lip-9.md

```diff
@@ -0,0 +1,95 @@
+---
+lip: 9
+title: Libra Private Transaction Metadata
+authors: Daniel Prinz
+status: Draft
+type: Informational
+created: 09/10/2020
+---
+# Summary
+This LIP describes the conceptual model and implementation of a private, on-chain description attribute, that could be added to every P2P transaction. An example of that would be short and descriptive information about the transaction, like what are the purchased items that are being paid in a transaction. 
+
+# Abstract / Overview
+The Libra blockchain can carry informational metadata with every transaction made. This data could vary and usually used for validating both sides of the transaction could "understand" its mean. While this metadata deals with the functioning of the sender and receiver VASPs, and/or with the FATF travel rule regulatory constraints, the transaction info itself remains opaque to the end-users. This property is missing from the current payment experience of any traditional network.
+This LIP goal is to propose a standard way of describing any transaction while keeping the following principals:
+1. Indisputable data
+2. Privacy of the end-user (sender)
+3. Ownership of the data by the end-user 
+
+
+# Terminology
+* *End users/Customers* are custodial or non-custodial wallet users which wish to transfer funds from their account to the beneficiary account (Receiver/Merchant)
+* *End-user wallet* is an on-chain VASP who provides custodian services to the end-users. This entity is considered optional and this role could be played by the end-users in a non-custodial wallet scenarios
+* *Receiver/Merchant VASP* is an entity that provides the same VASP services to the other side of the transaction, such as a regular receiver or merchant
+* *Receiver/Merchant* is the beneficiary of the transaction
+* *Private/Public keys* are keys used to sign and verify as well as encrypt and decrypt data by both sides. a private key is usually used to sign and encrypt while a public key is usually used for verification  
```

Comment:
Encrypting using classical public key cryptography and storing this data in a ledger (that will stay accessible for ever) can cause many issues in the future (from information disclosure to reputation).

Assume there is a cryptanalytic attack or some regulations in the future that demand a minimum key size or even post-quantum security. Unlike signatures where we can rotate keys in theory, publicly available encrypted data cannot be updated. Even if this was tolerable today (I doubt it is in practice), we don't know the situation in 10-20 years from now, so we have to be very careful with f we ever allow encryption on chain for potentially sensitive data.

Reply:
I agree that storing private data on-chain introduces a potential risk of having it exposed if a new crypto-attack will hit. 
Note that I don't suggest storing *personal* data by itself, rather the payment details (like description). the identity behind the data remains unknown to an outside observer.